src/dbnode/persist/fs/retriever_test.go

```diff
@@ -288,10 +360,48 @@ func testBlockRetrieverHighConcurrentSeeks(t *testing.T, shouldCacheShardIndices
 		}()
 	}
 
-	// Wait for all routines to be ready then start.
+	// Wait for all routines to be ready.
 	readyWg.Wait()
+	// Allow all the goroutines to begin.
 	startWg.Done()
 
+	// Setup concurrent block lease updates.
+	sem := make(chan struct{}, updateOpenLeaseConcurrency)
```

Comment:
Would it perhaps be more idiomatic to use a worker pool here to control concurrency instead of `sem` channel?  Had to read the code a little just to verify it was being used to bound concurrency, vs being used for some other mechanism.

Reply:
sure thing